File: Java/src/main/java/org/revian/MySet.java
package org.revian;

import java.util.ArrayList;
import java.util.List;

public class MySet<T> {
    private List<T> elements;

    public MySet() {
        elements = new ArrayList<>();
    }

    public void add(T element) {
        if (element == null) {
            throw new IllegalArgumentException("Element cannot be null");
        }
        if (!contains(element)) {
            elements.add(element);
        }
    }

    public void remove(T element) {
        if (element == null) {
            throw new IllegalArgumentException("Element cannot be null");
        }
        elements.remove(element);
    }

    public boolean contains(T element) {
        return elements.contains(element);
    }

    public int size() {
        return elements.size();
    }

    public boolean isEmpty() {
        return elements.isEmpty();
    }
}

File: CPP/src/MySet.cpp
//
// Created by Vivek Krishnan on 08/01/2025.
//

#include "../header/MySet.h"

// Check if an element exists in the set
bool MySet::contains(int element) const {
    for (const int& el : elements) {
        if (el == element) {
            return true;
        }
    }
    return false;
}

// Add an element to the set
void MySet::add(int element) {
    if (!contains(element)) {
        elements.push_back(element);
    }
}

// Remove an element from the set
void MySet::remove(int element) {
    for (auto it = elements.begin(); it != elements.end(); ++it) {
        if (*it == element) {
            elements.erase(it);
            return;
        }
    }
    throw std::out_of_range("Element not found in the set");
}

// Check if an element exists in the set
bool MySet::exists(int element) const {
    return contains(element);
}

// Get the number of elements in the set
int MySet::size() const {
    return elements.size();
}

// Check if the set is empty
bool MySet::isEmpty() const {
    return elements.empty();
}

// Get all elements of the set
std::vector<int> MySet::getElements() const {
    return elements;
}

// Clear all elements in the set
void MySet::clear() {
    elements.clear();
}
